backend/tarsy/agents/prompts/builders.py

```
@@ -0,0 +1,456 @@
+                
+                # Skip empty lines when not in a section
+                if not line and not current_section:
+                    continue
+                
+                # Check for stop conditions first
+                if self._should_stop_parsing(line):
+                    self._finalize_current_section(parsed, current_section, content_lines)
+                    break
+                
+                # Handle Final Answer (can appear at any time)
+                if self._is_section_header(line, 'final_answer', set()):  # Always allow final_answer
```

Ruff (0.12.2)

Line too long (101 > 88)

(E501)